File: single-language-analysis/dataset-files/python/test_smell_040.py
import pytest
import os
import tempfile

class TestFunctest:
    def test_smell_040(self):
        fake_json_data = "Deathcounter: 0"
        try:
            temp_file_path = tempfile.NamedTemporaryFile(delete=False, suffix=".json")
            with open(temp_file_path.name, 'w') as f:
                f.write(fake_json_data)

            result = func.read_json_from_file()

            assert 0 == result

            os.remove(temp_file_path.name)
        except IOError as e:
            print(e)

File: single-language-analysis/dataset-files/java/TestSmell040.java
// Original URL: https://github.com/MaximeAttevall/AdventureGame/blob/dd9997a20b2270a6c5dd80e995b442250bd64088/funcTest.java#L15-L36

public class TestSmell040 {
    @Test
    public void testReadJsonFromFile() {
        // Anta att JSON-filen finns och innehåller en giltig sträng
        String fakeJsonData = "Deathcounter: 0";
        try {
            // Skapa en temporär fil med testdata
            Path tempFilePath = Files.createTempFile("tempTestFile", ".json");
            Files.write(tempFilePath, fakeJsonData.getBytes());

            // Testa att läsa från filen
            int result = func.readJsonFromFile();

            // Kontrollerar att resultatet är 0 baserat på det falska JSON-data som skapades
            assertEquals(0, result);

            // Radera den temporära filen efter testet
            Files.delete(tempFilePath);
        } catch (IOException e) {
            // Fångar IOException om det uppstår
            e.printStackTrace();
        }
    }
}
